src/pvecontrol/sanitycheck/tests/vm.py

```diff
@@ -0,0 +1,39 @@
+from pvecontrol.vm import VmStatus
+from pvecontrol.sanitycheck.checks import Check, CheckCode, CheckType, CheckMessage
+
+
+class VmsStartOnBoot(Check):
+
+    id = "vms_start_on_boot"
+    type = CheckType.VM
+    name = "Check vms startonboot option"
+
+    def run(self):
+        vms = [vm for vm in self.proxmox.vms if vm.template == 0]
+        for vm in vms:
+            self._check_vm_statonboot_option(vm)
+
+    def _check_vm_statonboot_option(self, vm):
+        if self._vm_has_startonboot_enabled(vm):
+            self._check_vm_statonboot_enabled(vm)
+        else:
+            self._check_vm_statonboot_disabled(vm)
+
+    def _vm_has_startonboot_enabled(self, vm):
+        return vm.config.get("onboot") == 1
```

Comment:
The check is broken, When the Start on Boot option is defined as false in vm options, or maybe never changed via PVE UI the option `onboot` is not defined in vm option.
However the code handle this as a `onboot` == 1 because of the get() which always return true even if property does not exists.